### ISSUE TITLE
Reduce ambiguity in code writing

### ISSUE THREAD

**Issue by paulincai** [Meteor-Community-Packages/meteor-collection2#463]
https://github.com/Meteor-Community-Packages/meteor-collection2/blob/6b89fa5203d463133087181c4b17e0b2e7b5332d/package/collection2/main.js#L219

At line 219 and 220 among the parameters "args" is used in both functions and the initialization on line 220 does not use "const" or "let".
It is hard to follow the scope of various variables.

**Comment by jankapunkt** [comment]
Thank you @paulincai for reporting this. I checked the code and I think this is problematic in multiple ways.

- multiple scopes for same variable names can easily lead to unpredictable code flow
- the args are actually mutated within the functions, which can lead to flaky tests and also unpredictable code flow
- code is much less comprehensible
 I would go with a PR, trying to not be breaking

**Comment by StorytellerCZ** [comment]
Keeping this open for the next part of working on this as per @jankapunkt
